### PR TITLE
features: add use case for questions

### PR DESCRIPTION
**Pull request by PanVova**

### CODE REVIEW

File: api/src/main/java/com/panhuk/api/api/QuestionApi.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.panhuk.api.api
+
+import com.panhuk.api.dto.BaseResponse
+import kotlinx.coroutines.flow.Flow
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface QuestionApi {
+  @GET("api_token.php")
```
Comment (PanVova) [inline api/src/main/java/com/panhuk/api/api/QuestionApi.kt]: api.php

File: api/src/main/java/com/panhuk/api/di/questionApi/QuestionApiComponent.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    QuestionApiModule::class,
+    BaseApiModule::class
+  ]
+)
+interface QuestionApiComponent {
+  fun questionsApi(): QuestionApi
+
+  @Component.Builder
+  interface Builder {
+    fun questionsApiModule(module: QuestionApiModule): Builder
```
Comment (PanVova) [inline api/src/main/java/com/panhuk/api/di/questionApi/QuestionApiComponent.kt:18]: add baseApiModule for Builder

File: api/src/main/java/com/panhuk/api/di/sessionTokenApi/SessionTokenApiComponent.kt [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
-import com.panhuk.api.SessionTokenApi
+import com.panhuk.api.api.SessionTokenApi
+import com.panhuk.api.di.base.BaseApiModule
 import dagger.Component
 
 @Component(
   modules = [
-    SessionTokenApiModule::class
+    SessionTokenApiModule::class,
+    BaseApiModule::class
```
Comment (PanVova) [inline api/src/main/java/com/panhuk/api/di/sessionTokenApi/SessionTokenApiComponent.kt:10]: add BaseApi for Builder

File: dataSource/src/main/java/com/panhuk/datasource/QuestionReader.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.panhuk.datasource
+
+import kotlinx.coroutines.flow.Flow
+import model.Question
+
+interface QuestionReader {
```
Comment (PanVova) [inline dataSource/src/main/java/com/panhuk/datasource/QuestionReader.kt]: Rename to QuestionDSReader

Comment (PanVova) [reply dataSource/src/main/java/com/panhuk/datasource/QuestionReader.kt]: And change to thiis Flow<List<Question>?>

File: domain/src/main/java/model/Question.kt [hunk 5]
```diff
@@ -1,13 +1,9 @@
 package model
```
Comment (PanVova) [inline domain/src/main/java/model/Question.kt]: move to other package with the name panhuk...

File: dataSourceDI/src/main/java/com/panhuk/datasourcedi/di/QuestionComponent.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  dependencies = [
+    QuestionApiComponent::class
+  ],
+  modules = [
+    QuestionModule::class
+  ]
+)
+interface QuestionComponent {
+
+  @Api fun sessionTokenApiReader(): QuestionReader
```
Comment (PanVova) [inline dataSourceDI/src/main/java/com/panhuk/datasourcedi/di/QuestionComponent.kt]: change the name to QuestionApiReader

File: dataSourceDI/src/main/java/com/panhuk/datasourcedi/di/QuestionModule.kt [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package com.panhuk.datasourcedi.di
+
+import com.panhuk.api.api.QuestionApi
+import com.panhuk.api.api.SessionTokenApi
+import com.panhuk.datasource.QuestionReader
+import com.panhuk.datasource.SessionTokenCache
+import com.panhuk.datasource.SessionTokenReader
```
Comment (PanVova) [inline dataSourceDI/src/main/java/com/panhuk/datasourcedi/di/QuestionModule.kt]: delete unused imports

File: dataSourceImpl/src/main/java/com/panhuk/dataSourceimpl/QuestionApiDataSource.kt [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import model.Difficulty
+import model.Question
+import model.Type
+
+class QuestionApiDataSource(private val api: QuestionApi) : QuestionReader {
+  override val questions: Flow<List<Question?>>
+    get() = api.getQuestions().map {
+      it.results.map { questionDTO ->
+        Question(
+          category = Category.valueOf(questionDTO.category),
```
Comment (PanVova) [inline dataSourceImpl/src/main/java/com/panhuk/dataSourceimpl/QuestionApiDataSource.kt]: relocate the it.results... to the file mapper

File: repository/src/main/java/com/panhuk/repository/QuestionRepository.kt [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package com.panhuk.repository
+
+import kotlinx.coroutines.flow.Flow
+import model.Question
+
+interface QuestionRepository {
+  val questions: Flow<List<Question?>>
```
Comment (PanVova) [inline repository/src/main/java/com/panhuk/repository/QuestionRepository.kt]: val questions: Flow<List<Question>?>

File: repository/src/main/java/com/panhuk/repository/SessionTokenRepository.kt [hunk 10]
```diff
@@ -2,8 +2,7 @@ package com.panhuk.repository
 
 import kotlinx.coroutines.flow.Flow
 
-interface SessionTokenReader {
-
+interface SessionTokenRepository {
```
Comment (PanVova) [inline repository/src/main/java/com/panhuk/repository/SessionTokenRepository.kt]: rename SesssionTokenRepoReader

File: repositoryImpl/src/main/java/com/panhuk/repository_impl/QuestionRepoImpl.kt [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+import model.Question
+
+typealias RepositoryQuestionReader = QuestionRepository
+typealias DataSourceQuestionReader = com.panhuk.datasource.QuestionReader
+
+class QuestionRepoImpl(
+  private val questionApiReader: DataSourceQuestionReader
+) : RepositoryQuestionReader {
+
+  override val questions: Flow<List<Question?>>
```
Comment (PanVova) [inline repositoryImpl/src/main/java/com/panhuk/repository_impl/QuestionRepoImpl.kt]: Flow<List<Question>?>

File: useCase/src/main/java/com/panhuk/useCase/GetQuestionsUseCase.kt [hunk 12]
```diff
@@ -0,0 +1,8 @@
+package com.panhuk.useCase
+
+import model.Question
+import kotlinx.coroutines.flow.Flow
+
+interface GetQuestionsUseCase {
```
Comment (PanVova) [inline useCase/src/main/java/com/panhuk/useCase/GetQuestionsUseCase.kt:6]: Flow<List<Question?>>

File: usecaseImpl/src/main/java/com/panhuk/usecaseImpl/GetQuestionsUseCaseImpl.kt [hunk 13]
```diff
@@ -0,0 +1,13 @@
+package com.panhuk.usecaseImpl
+
+import com.panhuk.repository.QuestionRepository
+import com.panhuk.useCase.GetQuestionsUseCase
+import kotlinx.coroutines.flow.Flow
+import model.Question
+
+class GetQuestionsUseCaseImpl(private val questionRepository: QuestionRepository) :
+  GetQuestionsUseCase {
+  override fun getQuestions(): Flow<List<Question?>> {
```
Comment (PanVova) [inline usecaseImpl/src/main/java/com/panhuk/usecaseImpl/GetQuestionsUseCaseImpl.kt]: Flow<List<Question>?>

File: dataSourceImpl/src/main/java/com/panhuk/dataSourceimpl/QuestionApiDataSource.kt (hunk 8 above)
Comment (PanVova) [reply dataSourceImpl/src/main/java/com/panhuk/dataSourceimpl/QuestionApiDataSource.kt]: rename the incorrectAnswers to allAnswer so we need to add this list correct answer too

File: usecaseImpl/src/main/java/com/panhuk/usecaseImpl/GetQuestionsUseCaseImpl.kt (hunk 13 above)
Comment (PanVova) [reply usecaseImpl/src/main/java/com/panhuk/usecaseImpl/GetQuestionsUseCaseImpl.kt]: shuffle list

**Review by NickHuk — APPROVED** [review]

**Pull request merged**